lib/katello/tasks/upgrades/3.8/clear_checksum_type.rake

```diff
@@ -0,0 +1,23 @@
+namespace :katello do
+  namespace :upgrades do
+    namespace '3.8' do
+      desc "Clear checksum type for on-demand repositories"
+      task :clear_checksum_type => %w(environment) do
+        User.current = User.anonymous_admin
+
+        repos = Katello::Repository.on_demand.yum_type.where.not(checksum_type: nil, url: nil)
+        repos.update_all(checksum_type: nil)
+        repos.find_each do |repo|
+          begin
+            if repo.find_distributor[:config]&.delete(:checksum_type)
+              Katello.pulp_server.resources.repository.update_distributor(
+                repo.pulp_id, repo.find_distributor[:id], repo.find_distributor[:config])
+            end
+          # rubocop:disable HandleExceptions
+          rescue RestClient::ResourceNotFound
+          end
```

Comment:
I swear i commented this yesterday but i don't see my comment, shouldn't this update the source_repo_checksum_type to that of the scratchpad for all "Library repos" as well?

Reply:
Makes sense to me :)